madmom/audio/spectrogram.py

```diff
@@ -1235,3 +1238,106 @@ def process(self, data, **kwargs):
             data, crossover_frequencies=self.crossover_frequencies,
             fmin=self.fmin, fmax=self.fmax, norm_filters=self.norm_filters,
             unique_filters=self.unique_filters, **kwargs)
+
+
+class SemitoneBandpassSpectrogram(np.ndarray):
+    """
+    Construct a semitone spectrogram by using time domain filters as
+    described in [1]_.
+
+    Parameters
+    ----------
+    filename : str
+        Name of the audio file or file handle.
+    fps : int, optional
+        Desired sample rate of the signal [Hz].
+    midi_min : int, optional
+        Lowest frequency [MIDI note] of the spectrogram
+    midi_max : int, optional
+        Highest frequency [MIDI note] of the spectrogram
+
+    References
+    ----------
+    .. [1] Meinard Müller,
+            "Information retrieval for music and motion",
+            Berlin: Springer, 2007.
+
+    """
+
+    # pylint: disable=super-on-old-class
+    # pylint: disable=super-init-not-called
+    # pylint: disable=attribute-defined-outside-init
+
+    def __init__(self, filename, fps=50, midi_min=21, midi_max=108):
+        # this method is for documentation purposes only
+        pass
+
+    def __new__(cls, filename, fps=50, midi_min=21, midi_max=108):
+
+        # determine the name of the file if it is a file handle
+        if isinstance(filename, Signal):
+            raise NotImplementedError(
+                'Computing a SemitoneBandpassSpectrogram from a Signal '
+                'instance is not yet supported, because resampling a Signal '
+                'with scipy is much slower than loading the file and '
+                'resampling with ffmpeg/avconv')
```

Comment:
Ok, now I see why you used `filename` above. However, we should fix the resampling in the `Signal` class instead of raising an error here.